irohad/ordering/impl/ordering_gate_impl.hpp

```diff
@@ -76,26 +80,34 @@ namespace iroha {
       /**
        * Try to push proposal for next consensus round
        */
-      void tryNextRound();
+      void tryNextRound(
+          shared_model::interface::types::HeightType last_block_height_);
 
       rxcpp::subjects::subject<
           std::shared_ptr<shared_model::interface::Proposal>>
           proposals_;
+
+      rxcpp::subjects::subject<shared_model::interface::types::HeightType>
+          net_proposals_;
       std::shared_ptr<iroha::network::OrderingGateTransport> transport_;
 
-      /// invariant: true if proposal can be pushed to subscribers
-      std::atomic_bool unlock_next_{true};
+      std::mutex proposal_mutex_;
 
       /// queue with all proposals received from ordering service
       tbb::concurrent_priority_queue<
           std::shared_ptr<shared_model::interface::Proposal>,
           ProposalComparator>
           proposal_queue_;
 
+      // last commited block height
```

Comment:
`///` should be used, because `//` is simple comment while `///` is documentation, so that doxygen parses it correctly.